test/amperity/gocd/secret/vault/plugin_api.clj

```diff
@@ -0,0 +1,58 @@
+(ns amperity.gocd.secret.vault.plugin-api
+  (:require [clojure.test :refer :all]
+            [amperity.gocd.secret.vault.plugin :as plugin]
+            [amperity.gocd.secret.vault.util :as u])
+  (:import (com.thoughtworks.go.plugin.api.request DefaultGoPluginApiRequest)
+           (com.thoughtworks.go.plugin.api.response DefaultGoPluginApiResponse)))
+
+
+;; Utils ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(defn default-go-plugin-api-request
+  "A .DefaultGoPluginApiRequest with the given name and params"
+  ([request-name request-params]                            ;; string map -> .DefaultGoPluginApiRequest
```

Comment:
comments should ideally go into the docstring of the function, or if they are truly just for understand a line in a function, precede instead of printing on the same line.

Here it looks like you are annotating the inputs and outputs, which is great!

Instead, maybe a markdown list of arguments, their types and semantics. You can mention what this returns but also mark it with a return type directly if its useful for avoiding reflection in interop.


```clojure
(defn default-go-plugin-api-request
  "A .DefaultGoPluginApiRequest with the given name and params

   Parameters:
   - `request-name` string of the name of the request
   - `request-params` map of parameters to the request (ed: listing relevant keys here)"
  ^DefaultGoPluginApiRequest 
  ([request-name request-params]
   ,,,)
```

This way, the nice documentation you are trying to provide will show up in either auto-generated documentation and/or the result of the `doc` function for someone exploring the code from a repl.